### ISSUE TITLE
Pattern for handling `has_one` and `has_many` associations

### ISSUE THREAD

**Issue by jesenko** [thoughtbot/ex_machina#14]
Is there a way to handle creation of `has_one` and `has_many` associations? Namely, to have something like:

```elixir
def factory(:article) do
  %Article{
    title: "My Awesome Article",
    comments: %Comment{
      body: "I will be always here!"
    }
  }
end
```
and to create it 
```elixir
article_with_comment = create(:article)
```

Current, if I try to do this I get error  `... Assocs can only be manipulated via changesets, be it on insert, update or delete.` when `Repo.insert!` [is called](https://github.com/thoughtbot/ex_machina/blob/master/lib/ex_machina/ecto.ex#L90).

If instead model to insert is converted to changeset first, associations are inserted as well,
```elixir
%{__struct__: model} = record
struct(model) |> Ecto.Changeset.change(drop_ecto_fields(record)) |> repo.insert!
```

Not sure if this is the right approach though...

**Comment by paulcsmith** [comment]
Thanks for this issue and the PR, another way to handle this if you want to make things a bit more flexible is to use functions. Here's something we've been doing lately.

```elixir
def with_comment(article) do
  create(:comment, article: article)
  article
end

create(:article) |> with_comment
```

That way you can add comments only when you need them. It works great for creating join tables as well. I still agree we need a way to handle `has_one` and `has_many` associations though for cases where this isn't ideal

**Comment by jsteiner** [comment]
Closing this since #15 was merged into 0.4.0.
